src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node heap;
+    private Node tall;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<T>(null, value, null);
+            heap = newNode;
+            tall = newNode;
```

Comment:
this line is the same in `if` and `else` block, so let's move it after `if-else` construction